src/main/java/org/springframework/data/elasticsearch/core/SearchHitMapping.java

```diff
@@ -194,7 +195,7 @@ public void mapHitsInCompletionSuggestion(@Nullable Suggest suggest) {
 	 */
 	private SearchHits<?> mapInnerDocuments(SearchHits<SearchDocument> searchHits, Class<T> type) {
 
-		if (searchHits.getTotalHits() == 0) {
+		if (searchHits.isEmpty()) {
```

Comment:
Do you have a test case showing the error with the old code? `SearchHits.getTotalHits()` returns a primitive `long` value. How does this produce an `IndexOutOfBoundsException`? It might be a `NaN` but then we should rather fix the setting of the `totalHits`property.

Reply:
The `IndexOutOfBoundsException` is thrown in line 203 when accessing `searchHits.getSearchHit(0)`. You can tell Elastic not to return inner hits by setting the `size` in `inner_hits` to 0. This way you can keep track of total hits without returning any results.

Reply:
By the way, it's not so easy to add a test case for this, because the exceptions are caught here and only a warning is logged. I am not sure if we should change that as well.

Reply:
ok for the change. 

The exception should probably not be swallowed, it rather should be wrapped in a `UncategorizedElasticsearchException`.

As for the test: this could be done with a method in `org.springframework.data.elasticsearch.core.ElasticsearchTemplateTests`, insert a document, then do a search with the `inner_hits` set to zero which should then produce the exception with the old code.

Reply:
**searchHits.getTotalHits() == searchHits.getSearchHits().size()** ?
**_searchHits.getTotalHits()_** - number of found.
**_searchHits.getSearchHits()_** - number of received for current page.
They may not be equal. Or am I wrong?

maybe: 
```
if (!searchHits.hasSearchHits()) {
```

and **searchHits.getSearchHit(0)** not throw _IndexOutOfBoundsException_

Reply:
@sothawo everything has now been addressed and force-pushed.